### ISSUE TITLE
Error: "Template html does not match input"

### ISSUE THREAD

**Issue by bramaudi** [ryansolid/solid#405]
This error caused by missing `<thead>` in my `<table>` inside `<For>`, but without `<tbody>` is fine.

Is this solid a11y feature? If yes, would be better add missing element to error massage or something detailed about what the *missed* things.

Here's the example:
https://playground.solidjs.com/?hash=571184799

Nothing displayed even on console (playground) until you uncomment the `<thead>` tag.

**Comment by ryansolid** [comment]
This is a catch all sort of error. It can happen when people do misformed HTML stuff in JSX. There are a few others like putting divs in spans or certain elements in others that can't support them. Honestly I don't know them all so it seemed easiest to do a simple check on the elements as a safety precaution.

Because Solid clones templates and does walks I need the generated code to reflect the template and if you form non-valid html it will compile into the string fine but the browser will automatically fix it in the clone and it will basically break the app in really unexpected ways. So truthfully I don't know what's broken until the browser fixes it at runtime.

If there was a library that could validate html partials that I could run on the server as part of the babel plugin I'd be more than happy to but don't want to put anything more on the runtime. I'd be actually really happy to do this ahead of time. Just never found a library that did this and I honestly don't know all the rules myself.

**Comment by bramaudi** [comment]
Agree with "*... don't want to put anything more on the runtime*"

Table element are complicated, that's why i always write it with simple (and probably wrong) way just because browser will fixes it.

I am fine with this Solid behavior, just confuse about error message at first time, maybe add "... or html is misformed" to the message would be clear to me.
